### ISSUE TITLE
Improve handling of errors in connectors

### ISSUE THREAD

**Issue by reneaaron** [getAlby/lightning-browser-extension#2377]
### Feature description

Currently errors during account initialization are in some cases not handled, which can lead to some erroneous states in the extension. 

While there is a toast message informing the user that something went wrong, some parts of the UI do not reflect this change. 

**Example**
1. Unlock the extension
2. Choose account A
3. Switch to account B that throws an error during initialization (e.g. node is down, wrong credentials, etc)
4. The extension still displays the balance of the  account A

### Describe the solution

Errors should be handled and a message should be displayed to the user. @stackingsaunter Created some UI to reflect the failed loading in the extension. Feel free to add the designs here. :raised_hands: 

### Are you working on this?

No

**Comment by stackingsaunter** [comment]
<img width="1088" alt="Screenshot 2023-04-27 at 18 18 02" src="https://user-images.githubusercontent.com/68239231/234926394-a1bd01cd-988e-499c-826b-bc29e7b18a1f.png">

When "healthy" account is selected:

<img width="576" alt="Screenshot 2023-04-27 at 18 18 19" src="https://user-images.githubusercontent.com/68239231/234926448-485a1d7a-fca4-4b71-b3ad-d0176085a664.png">

**Comment by lujakob** [comment]
@reneaaron this builds upon the changes I suggested, right? I could give it a try.

**Comment by stackingsaunter** [comment]
Coming back what @rolznz said in #2499 

>I think it might be better to have a full error screen with a refresh button. Otherwise we need to individually block all the functionality of different buttons (send, receive, etc).

I changed copy a bit, added try again button, and maybe we can add a button that will lead straight to add a new account of the same connector as the node that user is trying to connect?

![image](https://github.com/getAlby/lightning-browser-extension/assets/68239231/fcb9d779-7dba-4762-9c05-031db734d846)

**Comment by rolznz** [comment]
@stackingsaunter the only issue I can see with the "add a new account" is that they probably still have to remove this one (if it is not usable). What about "Go to accounts" instead?

**Comment by reneaaron** [comment]
I don't think "add a new account" is particularly helpful at this place:

If your node is offline you'd want to fix the connection (or remove it if it's permanently offline). 

I'd just have the "try again" button there.

**Comment by stackingsaunter** [comment]
@reneaaron oftentimes I think the connection is broken because your node went offline and you need to set a new LNC session, so you need to add it again. But I am OK with going "Try again" only.

I think that's ready to go now! @lujakob 

Here's the head:
![Expression8](https://github.com/getAlby/lightning-browser-extension/assets/68239231/2627f07e-4b33-46d0-900b-34e6c8c29cc6)

![image](https://github.com/getAlby/lightning-browser-extension/assets/68239231/d873cbb9-1fc5-4f7b-829a-97a101487d3c)

**Comment by lujakob** [comment]
@stackingsaunter I want to wait for #2456 to be merged before I start, because I think it has some dependencies.

**Comment by lujakob** [comment]
@stackingsaunter @reneaaron I started working on this. I'm not sure if we can have the "error" state in red color for the list of accounts in the AccountMenu. The error state is not stored `state.getState().accounts` and I'm not sure if we should store it there. I'm also not sure if this is actually desired UX. To be reminded the last time I wanted to select an account it didn't work? What do you guys think?

**Comment by reneaaron** [comment]
@lujakob I agree. No need to save the connection state an accounts basis. I think it's good for a start to just show the error for the current active connector if that makes things easier?

**Comment by lujakob** [comment]
@reneaaron do you have a clue how to implement the alert icons? One with the rounded border the other one with the triangle? I did not find them in the bitcoin icons and could not add a border with CSS? We already use them in the extension?

**Comment by stackingsaunter** [comment]
@lujakob Triangle is just a text sign "⚠". The circled one is a "!" icon in bitcoin icons that you can add a round border I think (it was designed this way on purpose, not sure why, for faster loading?)

Also, please use red-600 on text, not red-500 for the text for better contrast

**Comment by lujakob** [comment]
> @lujakob Triangle is just a text sign "⚠". The circled one is a "!" icon in bitcoin icons that you can add a round border I think (it was designed this way on purpose, not sure why, for faster loading?)
> 
> Also, please use red-600 on text, not red-500 for the text for better contrast

Thx, I updated.

**Comment by stackingsaunter** [comment]
@lujakob sorry for messing around with colors AGAIN - I am now working on colors inconsistencies and solving those. Can you do red-700 for text and 300 in dark-mode?

<img width="670" alt="Screenshot 2023-07-03 at 22 32 09" src="https://github.com/getAlby/lightning-browser-extension/assets/68239231/1d0b719e-db99-4269-a8ac-8825c524a3c2">

Also color for warning surface would be red-50 light and red-900 dark 
<img width="609" alt="Screenshot 2023-07-03 at 22 33 02" src="https://github.com/getAlby/lightning-browser-extension/assets/68239231/1cd8a7aa-c2f7-44c9-8979-c7e2c903b61b">

Thanks 🙏

**Comment by lujakob** [comment]
@stackingsaunter I updated, but it looks a bit weird in dark mode, doesn't it?

<img width="380" alt="Bildschirmfoto 2023-07-05 um 06 34 32" src="https://github.com/getAlby/lightning-browser-extension/assets/886152/60977983-5a11-4d4f-85f7-f2e76f0e2576">

**Comment by stackingsaunter** [comment]
@lujakob I think you used warning (orange) instead of error (red) colors (text: red-700, dark: red-300 / bg red-50, dark: red-900)
It should look like this, although it looks brighter and more saturated on the screenshot than in Figma

![image](https://github.com/getAlby/lightning-browser-extension/assets/68239231/12fe831d-5fac-4a5a-a9d8-b3358c0fb0bf)
